src/game/common/audio/audioeventinfo.h

```diff
@@ -121,4 +123,72 @@ class AudioEventInfo : public MemoryPoolObject
 
 private:
     static FieldParse s_audioEventParseTable[];
+    friend class AudioEventRTS;
 };
+
+class DynamicAudioEventInfo : public AudioEventInfo
+{
+    IMPLEMENT_POOL(DynamicAudioEventInfo);
+
+    void *operator new(size_t size, void *dst) { return dst; }
+    void operator delete(void *p, void *q) {}
+
+public:
+    enum AudioOverrideFlags
+    {
+        OVERRIDE_AUDIO_NAME,
+        OVERRIDE_LOOP_FLAG,
+        OVERRIDE_LOOP_COUNT,
+        OVERRIDE_VOLUME,
+        OVERRIDE_MIN_VOLUME,
+        OVERRIDE_MIN_RANGE,
+        OVERRIDE_MAX_RANGE,
+        OVERRIDE_PRIORITY,
+        OVERRIDE_COUNT
+    };
+
+#ifdef GAME_DLL
+    DynamicAudioEventInfo *Hook_Ctor() { return new (this) DynamicAudioEventInfo(); }
+    DynamicAudioEventInfo *Hook_Ctor2(const AudioEventInfo &audio) { return new (this) DynamicAudioEventInfo(audio); }
+#endif
+
+    DynamicAudioEventInfo() {}
+    DynamicAudioEventInfo(const AudioEventInfo &audio) : AudioEventInfo(audio) {}
+
+    virtual ~DynamicAudioEventInfo() override {}
+    virtual bool Is_Level_Specific() override { return true; }
```

Comment:
This function is not `const`? Name suggests it should.

Reply:
fixed.